### PR TITLE
initial infra changes

### PR DESCRIPTION
**Pull request by akachwala-ch**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+build:
+	mvn versions:set -DnewVersion=$(version) -DgenerateBackupPoms=false
+	mvn package -DskipTests=true
+	cp ./target/$(artifact_name)-$(version).jar ./$(artifact_name).jar
+
+.PHONY: test
+test: test-unit
+
+.PHONY: test-unit
+test-unit: clean
```
[the remark this answers is not in the document]
Comment (Ngriffiths1) [reply Makefile:30]: There is also an option to change to this, but whatever preference you take is fine. 

```
.PHONY: test-unit
test-unit: clean
    mvn test -DexcludedGroups="integration-test"

.PHONY: test-integration
test-integration: clean
    mvn test -Dgroups="unit-test, integration-test"
```

File: Tiltfile.dev [hunk 2]
```diff
@@ -0,0 +1,17 @@
+local_resource(
+  name = 'dev:acsp-api',
+  cmd = 'mvn clean compile',
+  deps = ['src']
+)
+custom_build(
+  ref = '169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/acsp-api',
```
Comment (Ngriffiths1) [inline Tiltfile.dev:7]: Not sure if this line should be changed to reflect the correct region. Can you find out if this should be changed to be 

`eu-west-2`

File: Tiltfile.dev [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  command = 'mvn clean compile jib:dockerBuild -Dimage=$EXPECTED_REF',
+  live_update = [
+    sync(
+      local_path = './target/classes',
+      remote_path = '/app/classes'
+    ),
+    restart_container()
+  ],
+  deps = ['./target/classes']
+)
```
Comment (Ngriffiths1) [inline Tiltfile.dev:17]: Also might be worth checking if you need the lines which reference this within this file. 

https://github.com/companieshouse/presenter-account-api/blob/main/Tiltfile.dev#L20-#L26

File: terraform/groups/ecs-service/vault.tf [hunk 4]
```diff
@@ -0,0 +1,19 @@
+variable "hashicorp_vault_username" {
```
Comment (Ngriffiths1) [inline terraform/groups/ecs-service/vault.tf:1]: Can this file be removed and symlinked, similar to this request on a different repo https://github.com/companieshouse/identity-verification-api/pull/15#discussion_r1388030835

**Review by Ngriffiths1 — CHANGES_REQUESTED** [review]
Minor changes requested and comments raised

File: routes.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+app_name: acsp-api
```
[the remark this answers is not in the document]
Comment (akachwala-ch) [reply routes.yaml]: @Ngriffiths1 file removed.

File: ecs-image-build/docker_start.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+#
+# Start script for acsp-api
+
+PORT=18642
```
[the remark this answers is not in the document]
Comment (akachwala-ch) [reply ecs-image-build/docker_start.sh:5]: @Ngriffiths1 I have used this port based on the port selected by devs, is that still ok?
Script does have executable permissions. 
[](https://github.com/companieshouse/acsp-api/blob/main/src/main/resources/application.yaml)

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (akachwala-ch) [reply Makefile]: @Ngriffiths1 updated and I have gone with the first option.

File: Tiltfile.dev (hunk 2 above)
Comment (akachwala-ch) [reply Tiltfile.dev]: updated to '416670754337.dkr.ecr.eu-west-2.amazonaws.com/acsp-api'

File: Tiltfile.dev (hunk 3 above)
Comment (akachwala-ch) [reply Tiltfile.dev:17]: @Ngriffiths1 ran this via lead dev, toggle not need at present hence not included.

File: terraform/groups/ecs-service/vault.tf (hunk 4 above)
Comment (akachwala-ch) [reply terraform/groups/ecs-service/vault.tf]: @Ngriffiths1 symlinked, Thank you.

File: ecs-image-build/docker_start.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (Ngriffiths1) [reply ecs-image-build/docker_start.sh:5]: From that side of things it would not matter as it being run on ECS but happy to go with your value

**Review by Ngriffiths1 — APPROVED** [review]
LGTM

**Review by sampankumar — APPROVED** [review]

**Pull request merged**